CriteoPublisherSdk/Tests/UnitTests/DataConsent/CR_GdprTests.m

```diff
@@ -202,6 +202,46 @@ - (void)testPublisherRestrictionsMWithConsentStringForTcf2_0 {
                  CR_GdprTcfPublisherRestrictionTypeNone);
 }
 
+#pragma mark - Vendor consent
+
+- (void)testVendorConsentSimpleCases {
+#define AssertVendorConsent(tcfVersion, consents, expected)       \
+  do {                                                            \
+    [self.userDefaults clearGdpr];                                \
+    [self.userDefaults setGdprTcf##tcfVersion##GdprApplies:@YES]; \
+    [self.userDefaults setGdprTcf2_0VendorConsents:consents];     \
+    XCTAssertEqual([self.gdpr isVendorConsentGiven], expected);   \
+  } while (0);
+
+  AssertVendorConsent(1_1, nil, YES);
+  AssertVendorConsent(2_0, @"", YES);
+  AssertVendorConsent(2_0, @"malformed", YES);
+  AssertVendorConsent(
+      2_0,
+      @"---------+---------+---------+---------+---------+---------+---------+---------+---------+X--------+",
```

Comment:
Isn't it possible to execute the trick you've done in the commit before ? Formatting an int on `¨%091d` ?

Reply:
It is, it is just that I didn't want to trick myself with tricks everywhere and wanted to be explicit at least once :)
I'll edit this.

Reply:
Oh also it was because you cannot apply this trick to `X` but nevermind ;p